lib/recurly/resource.rb

```diff
@@ -824,15 +824,21 @@ def to_xml(options = {})
 
         # Duck-typing here is problematic because of ActiveSupport's #to_xml.
         case value
-        when Resource, Subscription::AddOns
+        when Resource
           value.to_xml options.merge(:builder => node)
-        when Array
+        when Array, Subscription::AddOns
           value.each do |e|
             if e.is_a? Recurly::Resource
-              # create a node to hold this resource
-              e_node = node.add_element Helper.singularize(key)
-              # serialize the resource into this node
-              e.to_xml(options.merge(builder: e_node))
+              if self.is_a? Recurly::SubscriptionAddOn
```

Comment:
I think we'll need to setup a call to go over this.

Reply:
Definitely. There's a lot to discuss!